src/generic/armcm_reset.c

```diff
@@ -9,10 +9,16 @@
 #include "board/irq.h" // irq_disable
 #include "board/misc.h" // try_request_canboot
 
+#if CONFIG_MACH_RP2040
+#define BOOTLOADER_START (CONFIG_RP2040_BOOTLOADER_START)
```

Comment:
User bootloader should be placed just after boot2 block at START_FLASH+256 address. Simple CONFIG_FLASH_START & 0xFF000000 do not work here.